core/src/databases.rs

```diff
@@ -0,0 +1,37 @@
+use crate::conn::MongoConnection;
+use crate::stmt::MongoStatement;
+use bson::Bson;
+use std::error::Error;
+
+#[derive(Debug)]
+pub struct MongoDatabases {
+    // The list of all the databases
+    databases_names: Vec<String>,
+    // The current database index.
+    current_row_index: u32,
+}
+
+// Statement for SQLTables(SQL_ALL_CATALOGS, "","").
+impl MongoDatabases {
+    // Create a new MongoStatement to list all the valid catalogs.
+    // Correspond to SQLTables(SQL_ALL_CATALOGS, "","").
+    // All columns except the TABLE_CAT column contain NULLs.
+    // The query timeout comes from the statement attribute SQL_ATTR_QUERY_TIMEOUT. If there is a
+    // timeout, the query must finish before the timeout or an error is returned.
```

Comment:
[nit] I think we should consistently call these `databases` instead of `catalogs`.

Reply:
I guess you wanted to highlight the previous line as this line is an ODBC call and SQL_ALL_CATALOGS is the catalog name filter.
Agreed. Let's be consistent.

Reply:
Done.